### PR TITLE
[AJ-1296] Organize data tables UI part 4

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/AJ-1296

Continuing to organize the UI code for data tables to separate functionality specific to Entity Service or WDS.

This moves SavedColumnSettings, EntitiesContent, DataTable, WDSContent, WdsTroubleshooter, and Data into the src/data folder.

Each commit moves one component at a time. The build process step provides good guard rails for updating imports. If an import has been missed, the build will fail when it can't resolve the module.

### CODE REVIEW

File: src/data/data-table/shared/DataTable.js [hunk 1]
```diff
@@ -42,6 +30,19 @@ import * as StateHistory from 'src/libs/state-history';
 import * as Style from 'src/libs/style';
 import * as Utils from 'src/libs/utils';
 
+import { concatenateAttributeNames } from '../entity-service/attribute-utils';
```
Comment (nawatts) [inline src/data/data-table/shared/DataTable.js:34]: I know it's unconventional in Terra UI to use relative imports outside the directory, but separating these helps highlight where components need to be updated to remove the assumption of using Entity Service.

**Review by calypsomatic — APPROVED** [review]

File: src/data/data-table/shared/DataTable.js (hunk 1 above)
Comment (jladieu) [reply src/data/data-table/shared/DataTable.js:34]: This comment might deserve to be in the code itself to explain :)

But just so I understand...is the idea to use a relative import to make an easy to grep string to help us keep track of (and maybe ultimately eliminate) this dependency?

**Review by jladieu — APPROVED** [review]

File: src/data/data-table/shared/DataTable.js (hunk 1 above)
Comment (nawatts) [reply src/data/data-table/shared/DataTable.js:34]: Yes. I was thinking more "visually clear" than "easy to grep", but same idea.

Comment (nawatts) [reply src/data/data-table/shared/DataTable.js:34]: Added a comment in https://github.com/DataBiosphere/terra-ui/pull/4178/commits/d113cb121ca482d44c7a8f0736089d01e0baffc8.

**Review by jladieu — APPROVED** [review]

**Pull request merged**